vdom/src/com/vdom/core/CardImplIntrigue.java

```diff
@@ -299,6 +422,84 @@ private void pawn(Game game, MoveContext context, Player currentPlayer) {
         }
     }
 
+    private void replace(Game game, MoveContext context, Player currentPlayer) {
+        ArrayList<Player> playersToAttack = new ArrayList<Player>();
+        for (Player player : game.getPlayersInTurnOrder()) {
+            if (player != currentPlayer && !Util.isDefendedFromAttack(game, player, this)) {
+                playersToAttack.add(player);
+                player.attacked(this.controlCard, context);
+            }
+        }
+
+        if(currentPlayer.getHand().size() <= 0) {
+            return;
+        }
+
+        Card cardToTrash = currentPlayer.controlPlayer.replace_cardToTrash(context);
+
+        if (cardToTrash == null) {
+            Util.playerError(currentPlayer, "Remodel did not return a card to trash, trashing random card.");
+            cardToTrash = Util.randomCard(currentPlayer.getHand());
+        }
+
+        int cost = -1;
+        int debt = -1;
+        boolean potion = false;
+        for (int i = 0; i < currentPlayer.hand.size(); i++) {
+            Card playersCard = currentPlayer.hand.get(i);
+            if (playersCard.equals(cardToTrash)) {
+                cost = playersCard.getCost(context);
+                debt = playersCard.getDebtCost(context);
+                potion = playersCard.costPotion();
+                playersCard = currentPlayer.hand.remove(i);
+
+                currentPlayer.trash(playersCard, this.controlCard, context);
+                break;
+            }
+        }
+
+        if (cost == -1) {
+            Util.playerError(currentPlayer, "Replace returned invalid card, ignoring.");
```

Comment:
At this point the card was already checked and another card from the hand was randomly chosen. Not much more one can do. Remodel (which is also not optional (before errata)) handles it the same way.